libraries/AP_Landing/AP_Landing_Deepstall.cpp

```diff
@@ -192,10 +192,24 @@ bool AP_Landing_Deepstall::verify_land(const Location &prev_WP_loc, Location &ne
         stage = DEEPSTALL_STAGE_WAIT_FOR_BREAKOUT;
         //compute optimal path for landing
         build_approach_path();
+        loiter_sum_cd = 0; // reset the loiter counter
         // no break
         }
     case DEEPSTALL_STAGE_WAIT_FOR_BREAKOUT:
         if (!verify_breakout(current_loc, arc_entry, height)) {
+            int32_t target_bearing = landing.nav_controller->target_bearing_cd();
+            int32_t delta = wrap_180_cd(target_bearing - last_target_bearing);
+            if (delta > 0) { // only accumulate turns in the correct direction
+                loiter_sum_cd += delta;
+            }
+            last_target_bearing = target_bearing;
+            // rebuild the approach path if we have done less then a full circle to allow it to be
+            // more into the wind as the estimator continues to refine itself, and allow better
+            // compensation on windy days. This is limited to a single full circle though, as on
+            // a no wind day you could be in this loop forever otherwise.
+            if (loiter_sum_cd < 36000) {
+                build_approach_path();
```

Comment:
It's a bit of expensive math to do twice in a row, there isn't any other side effect. I don't want to add an extra bool for tracking this edge case, I think I can flip some of the if's around though so that we only do the math once. Let me try that real quick.